using System;

namespace SquareRoot
{
    class Program
    {
        public static double Sqrt(double value)
        {
            if (value < 0)
                throw new System.ArgumentOutOfRangeException("value",
                   "Sqrt for negative numbers is undefined!");
            return Math.Sqrt(value);
        }

        static void Main(string[] args)
        {
           //int a = int.Parse(Console.ReadLine());

            try
            {
                Sqrt(-1);
            }
            catch (ArgumentOutOfRangeException ex)
            {
                Console.Error.WriteLine("Error: " + ex.Message);
               
            }
            catch(FormatException ex)
            {
                Console.WriteLine("InvalidNumber!");
                throw ex;
            } // в момента няма смисъл от това, тъй като за try-а се изисква да си е в правилния формат, за да провери горния метод
            finally
            {
                Console.WriteLine("Good bye.");
            }
           // Console.WriteLine(Math.Sqrt(a));
        }
    }
}
